shared/utility/motion/InverseKinematics.cpp

```diff
@@ -226,6 +226,152 @@ namespace motion {
             return positions;
         }
 
+        std::vector<std::pair<ServoID, double>> calculateLegJoints(const KinematicsModel& model,
+                                                                   const Eigen::Affine3d& target_,
+                                                                   const LimbID& limb) {
+            const double LENGTH_BETWEEN_LEGS             = model.leg.LENGTH_BETWEEN_LEGS;
+            const double DISTANCE_FROM_BODY_TO_HIP_JOINT = model.leg.HIP_OFFSET_Z;
+            const double HIP_OFFSET_X                    = model.leg.HIP_OFFSET_X;
+            const double UPPER_LEG_LENGTH                = model.leg.UPPER_LEG_LENGTH;
+            const double LOWER_LEG_LENGTH                = model.leg.LOWER_LEG_LENGTH;
+
+            std::vector<std::pair<ServoID, double>> positions;
+
+            double hipYaw     = 0;
+            double hipRoll    = 0;
+            double hipPitch   = 0;
+            double knee       = 0;
+            double anklePitch = 0;
+            double ankleRoll  = 0;
+
+            // Correct for input referencing the bottom of the foot
+            Eigen::Affine3d target(target_);
+            target = target.translate(Eigen::Vector3d(0.0, 0.0, model.leg.FOOT_HEIGHT));
+
+            // TODO remove this. It was due to wrong convention use
+            Eigen::Matrix4d Tci;
```

Comment:
What is this?

Reply:
A transformation (not necessarily homogeneous) from input coordinates to calc coordinates. I wanted a much shorter name than what it used to be

Reply:
I suppose that's fair. But is the comment still important?

Reply:
Yes, apparently we are still using the wrong conventions, thats what the transform is for lol